pwnlib/util/packing.py

```diff
@@ -756,15 +762,16 @@ def flat(*args, **kwargs):
     preprocessor = kwargs.pop('preprocessor', lambda x: None)
     filler       = kwargs.pop('filler', cyclic.de_bruijn())
     length       = kwargs.pop('length', None)
+    stacklevel   = kwargs.pop('stacklevel', 0)
 
-    if isinstance(filler, str):
-        filler = bytearray(six.ensure_binary(filler))
+    if isinstance(filler, (str, six.text_type)):
+        filler = bytearray(context._need_bytes(filler))
 
     if kwargs != {}:
         raise TypeError("flat() does not support argument %r" % kwargs.popitem()[0])
 
     filler = iters.cycle(filler)
-    out = _flat(args, preprocessor, make_packer(), filler)
+    out = _flat(args, preprocessor, make_packer(), filler, stacklevel + 2)
```

Comment:
Why +2 here?

Reply:
One extra stack level is used by the wrapper function returned by the decorator, tested